tsh.cc

```diff
@@ -293,48 +295,32 @@ void eval(char *cmdline)
 // string comparisons; however, the do_bgfg routine will need 
 // to use the argv array as well to look for a job number.
 //
-int builtin_cmd(char **argv) 
+int builtin_cmd(char **argv)
 {
-  string cmd(argv[0]);
-
- //cout << argv[0] << endl;
-
-
- 
-  //Check to see if the user wants to exit the shell
-  //if the first command is "quit" exit the tiny shell.
-  if(strcmp("quit", argv[0]) == 0)
-  {
-	  exit(0);
-  }
-
-  if(strcmp("jobs", argv[0]) == 0)
-  {
-
-	  //show the stored jobs in the job holder
-	  while(!jobHolder.empty())
-	  {
-		  //create a variable of type child struct to current child
-		  struct child mychild  = jobHolder.front();
-
-
-		  //no endl with the last cout statement
-		  //the captured command includes user captured carriage return
-		  cout << "[" << mychild.jobID <<"] ";
-		  cout << "(" << mychild.pid << ") ";
-		  cout << mychild.state << " ";
-		  cout << mychild.command;
-
-		  jobHolder.pop();
-	  }	  
-	  return 1;
-  }
+    string cmd(argv[0]);
 
 
+    if (strcmp(argv[0], "quit") == 0 || strcmp(argv[0], "q") == 0)
+    {
+        printf("Quitting...\n");
+        exit(0);
+    }
 
+    if (strcmp(argv[0], "jobs") == 0 || strcmp(argv[0], "j") == 0)
+    {
+        for (child* c : jobHolder)
```

Comment:
Here we can iterate over all the jobs using the vector like characteristics of a deque. Your previous implementation would destroy your job queue in the act of viewing each element.